pkg/vaulturl/helpers.go

```diff
@@ -0,0 +1,54 @@
+package vaulturl
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/jenkins-x/jx/pkg/util"
+	"github.com/pkg/errors"
+)
+
+var vaultURIRegex = regexp.MustCompile(`vault:[-_\w\/:]*`)
+
+// ReplaceURIs will replace any vault: URIs in a string, using the vault client
+func ReplaceURIs(s string, client Client) (string, error) {
```

Comment:
I would rather make this a method part of `secreturl.Client`, this will allow us to extend it with other URI schemes (e.g. k8s secret).

Reply:
the main reason I kept it here for now was to avoid touching `vault.Client` in case it interfered with your refactor ;).  Maybe we add a `ReplaceURIs` function to the `secreturl.Client` which for vault + localvault can reuse the same code + URL scheme - then do something different for k8s?

Reply:
how about we support `secret:` for either vault or local and `vault:` for only vault and `local:` for only local? then if folks want to be generic they use `secret:` URLs?

Reply:
we could add other URLs for vault - e.g. to use a specific named vault too?

Reply:
'secret:` sounds better to me in this case